tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -1434,34 +1431,30 @@ const reducer = createReducer(
       tableEditorSelectedTab: tab,
     };
   }),
-  on(actions.dataTableColumnEdited, (state, {dataTableMode, headers}) => {
-    const enabledNewHeaders: ColumnHeader[] = [];
-    const disabledNewHeaders: ColumnHeader[] = [];
-
-    // All enabled headers appear above all disabled headers.
-    headers.forEach((header) => {
-      if (header.enabled) {
-        enabledNewHeaders.push(header);
-      } else {
-        disabledNewHeaders.push(header);
+  on(
+    actions.dataTableColumnEdited,
+    (state, {source, destination, side, dataTableMode}) => {
+      let headers =
+        dataTableMode === DataTableMode.RANGE
+          ? [...state.rangeSelectionHeaders]
```

Comment:
To clarify:
- `dataTableColumnEdited` is currently only used for reordering
- runs and hparams features already define separate actions for reordering and adding (e.g. `runsTableHeaderAdded`, `runsTableHeaderOrderChanged`)

To prevent confusion (and for consistency) I'll change `dataTableColumnEdited` -> `dataTableColumnOrderChanged`

Reply:
(a separate `dataTableColumnAdded` action will be added in an upcoming PR)

Reply:
In that case can you `headers` from the props?

Reply:
Wait sorry - I need to correct myself. I _did_ add a `dataTableColumnAdded` in #6737, but it's unused because we don't actually have a feature for adding standard columns, only hparam columns. Thus, only the hparams action `dashboardHparamColumnAdded` will be used to add new columns.

Regarding passing header to the props, I propose that an action signature like [dashboardHparamColumnAdded](https://github.com/tensorflow/tensorboard/blob/4d7e337c9ad28cde11fbd8df981a9a235a27e2fc/tensorboard/webapp/hparams/_redux/hparams_reducers.ts#L93), i.e. only specifying {column, nextTo, side} and letting the reducer figure out how put it in the right place, is more in line with redux best practices, which advocate [putting logic in reducers, not components](https://redux.js.org/style-guide/#put-as-much-logic-as-possible-in-reducers); passing in headers via the action implies that the component must do some data manipulation first.

(Note that the previously implemented [runs table add action](https://github.com/tensorflow/tensorboard/blob/4d7e337c9ad28cde11fbd8df981a9a235a27e2fc/tensorboard/webapp/runs/store/runs_reducers.ts#L436) already uses the "put logic in reducers" pattern, albeit with a slightly different signature.)

Reply:
I fully agree with everything in this statement.
The thing that I find confusing is that the action [dataTableColumnEdited](https://github.com/tensorflow/tensorboard/blob/4d7e337c9ad28cde11fbd8df981a9a235a27e2fc/tensorboard/webapp/metrics/actions/index.ts#L237) takes [HeaderEditInfo](https://github.com/tensorflow/tensorboard/blob/master/tensorboard/webapp/metrics/internal_types.ts#L97) as props which has an unused attribute `headers`